src/components/common/CitiesList/CitiesList.jsx

```diff
@@ -1,21 +1,34 @@
 import CityCard from '../CityCard/CityCard';
 import { memo } from 'react';
+import usePopularCities from '../../../hooks/usePopularCities';
+import NotFoundSection from '../../../pages/CitiesPage/NotFoundSection';
+import CitiesListSkeleton from './CitiesListSkeleton';
+
+const CitiesList = ({ limit, increaseLimit, showLoadMoreButton }) => {
+  const { data: cities, error, isLoading } = usePopularCities();
 
-const CitiesList = ({ cities, limit, increaseLimit, showLoadMoreButton }) => {
   const defaultLimit = 10;
   const citiesLimit = limit || defaultLimit;
-  const displayedCities = cities.slice(0, citiesLimit);
-  const allCitiesDisplayed = citiesLimit >= cities.length;
+  const displayedCities = cities?.slice(0, citiesLimit) || [];
```

Comment:
having default value of data inside the hook will help to avoid all these fallbacks

Reply:
and let's wrap `displayedCities` into `useMemo` to avoid getting a new array instance